### PR TITLE
Add italic font when describing what to do next

### PR DESCRIPTION
**Pull request by cdrage**
**What kind of PR is this?**

/kind code refactorting
/kind docs
/kind enhancement

**What does does this PR do / why we need it**:

This PR:
 - Adds italic font when outputting what to "do next". Essentially, when
 using log.Info it's for a "bold" or "title" log. When using log.Italic
 it's mainly used for when we want to say to the user what to do next,
 such as using `odo push` to propagate changes
 - Removes random periods which aren't suppose to be there (all output
 should not be sentences with periods)

**Which issue(s) this PR fixes**:

N/A

**How to test changes / Special notes to the reviewer**:

Run: `odo create nodejs` or any other similar commands, such as `odo
url create` which have "next steps" in them.

Signed-off-by: Charlie Drage <charlie@charliedrage.com>

### CODE REVIEW

**Comment by cdrage** [comment]
Example: 
Screenshot from 2019-11-01 16-19-44

File: pkg/odo/cli/component/create.go [hunk 1]
```diff
@@ -466,7 +466,7 @@ func (co *CreateOptions) Validate() (err error) {
 	}
 
 	if !supported {
-		log.Infof("Warning: %s is not fully supported by odo, and it is not guaranteed to work.", *co.componentSettings.Type)
+		log.Infof("Warning: %s is not fully supported by odo, and it is not guaranteed to work", *co.componentSettings.Type)
```
Comment (mik-dass) [inline pkg/odo/cli/component/create.go]: https://travis-ci.com/openshift/odo/jobs/251973014#L9431
Please add the `.` at the end or remove it from the test

File: pkg/odo/cli/preference/set.go [hunk 2]
```diff
@@ -65,7 +65,7 @@ func (o *SetOptions) Run() (err error) {
 	if !o.configForceFlag {
 		if isSet := cfg.IsSet(o.paramName); isSet {
 			if !ui.Proceed(fmt.Sprintf("%v is already set. Do you want to override it in the config", o.paramName)) {
-				log.Info("Aborted by the user.")
+				log.Info("Aborted by the user")
```
Comment (dharmit) [inline pkg/odo/cli/preference/set.go]: Should this one not use `log.Italic` as well? We did use it for the case where app deletion was aborted.

Comment (cdrage) [reply pkg/odo/cli/preference/set.go]: No, anything Italic IMO should only be regarding "what to do next". for example, using `odo push`, etc.

File: pkg/odo/cli/application/delete.go [hunk 3]
```diff
@@ -83,9 +83,9 @@ func (o *DeleteOptions) Run() (err error) {
 		if err != nil {
 			return err
 		}
-		log.Infof("Deleted application: %s from project: %v", o.appName, o.Project)
+		log.Italicf("\nDeleted application: %s from project: %v", o.appName, o.Project)
```
[the remark this answers is not in the document]
Comment (cdrage) [reply pkg/odo/cli/application/delete.go]: Follows the same format as the other titles, so going to keep using `:`

**Comment by mik-dass** [comment]
```
 should successfully execute list along with machine readable output [It]
    /go/src/github.com/openshift/odo/tests/integration/project/cmd_project_test.go:42
    Expected
      
        <string>: ACTIVE     NAME
                   dcpdqaecaw
                   myproject
                   yuwvbvjzyz
        
    to contain substring
        <string>: ugkhuaeqwg
```
/retest

**Comment by girishramnani** [comment]
4.3 intg is failing due to https://github.com/openshift/odo/issues/2357 @cdrage

File: pkg/odo/cli/application/delete.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (cdrage) [reply pkg/odo/cli/application/delete.go]: Ah yeah, you're right. This shouldn't be italic.

File: pkg/odo/cli/application/list.go [hunk 4]
```diff
@@ -97,8 +97,7 @@ func (o *ListOptions) Run() (err error) {
 			}
 			fmt.Println(string(out))
 		} else {
-
-			log.Infof("There are no applications deployed in the project '%v'.", o.Project)
+			log.Italicf("\nThere are no applications deployed in the project '%v'", o.Project)
```
Comment (mik-dass) [inline pkg/odo/cli/application/list.go]: This shouldn't be in italic too, it's not indicating next steps

**Comment by cdrage** [comment]
Updated! Thanks @mik-dass

File: pkg/odo/cli/url/create.go [hunk 5]
```diff
@@ -99,7 +99,7 @@ func (o *URLCreateOptions) Run() (err error) {
 		return errors.Wrapf(err, "failed to persist the component settings to config file")
 	}
 	log.Successf("URL %s created for component: %v\n", o.urlName, o.Component())
-	log.Info("To create URL on the OpenShift Cluster, please use `odo push`")
+	log.Italic("\nTo create URL on the OpenShift Cluster, please use `odo push`")
```
Comment (mik-dass) [inline pkg/odo/cli/url/create.go]: Please remove the new line, it's adding two newlines in the final output

```
[mrinaldas@localhost ~]$ odo url create example
 ✓  URL example created for component: nodejs-mrinaldas-axlj


To create URL on the OpenShift Cluster, please use `odo push`
```

Comment (cdrage) [reply pkg/odo/cli/url/create.go]: It's because of the previous success message having a newline :( I removed it from line 101.

**Pull request merged**